response.go

```diff
@@ -70,19 +81,19 @@ func (err connectionError) Timeout() bool {
 	return true
 }
 
-// serverError represents internal server errors.
+// ServerError represents internal server errors.
 // Implements `net.Error` interface.
-type serverError string
+type ServerError string
```

Comment:
It's not really my use case, I just went through it and was quickly changing things to be "idiomatic". I missed the `connectionError` type too, but it should have the same issue. Anyways, it's not really a problem if you don't think this should be exported, but it doesn't hurt anything to have the ServerError as an exported type.